docs/docs/guides/atom-effects.md

```diff
@@ -239,6 +241,89 @@ const currentUserIDState = atom({
 });
 ```
 
+## Asynchronous Storage Persistence
+
+If your persisted data needs to be retrieved asynchronously, you can either use a [`Promise`](#promise) in the `setSelf()` function or create an [asynchronous](#asynchronous) function and call that.
+
+### `Promise`
+
+By synchronously calling `setSelf()` with a `Promise` you'll be able to wrap the components inside of the `<RecoilRoot/>` with a `<Suspense/>` component to show a fallback while waiting for `Recoil` to load the persisted values.
+
+The `<Suspense/>` fallback will be shown until the `setSelf()` function has been called.
```

Comment:
"`<Suspense>` will show a fallback until the `Promise` provided to `setSelf()` resolves."

Reply:
Resolved.